### DISCUSSION TITLE
About debugging STM32F103C8T6 with STLink

### DISCUSSION THREAD

**Discussion by JingtaiChen** [stm32duino/Arduino_Core_STM32#2513] [General]
Hello, I use STLink SWDIO and SWCLK link F103C8T6, then click Debug,
<img width="34" alt="image" src="https://github.com/user-attachments/assets/9917c987-4f63-4a04-a5c1-095e27bfc29a">
Then the following fault is reported, at this time there is no problem in selecting SWD download, but it cannot be debugged, what should I do?
Failed to launch GDB: Error erasing flash with vFlashErase packet (from target-download)

Waiting for gdb server to start...[2024-09-11T18:25:20.559Z] SERVER CONSOLE DEBUG: onBackendConnect: gdb-server session connected. You can switch to "DEBUG CONSOLE" to see GDB interactions.
/Users/chenjingtai/Library/Arduino15/packages/STMicroelectronics/tools/xpack-openocd/0.12.0-1/bin/openocd -c "gdb_port 50000" -c "tcl_port 50001" -c "telnet_port 50002" -s /Users/chenjingtai/Desktop/sketch_sep11a -f "/Applications/Arduino IDE.app/Contents/Resources/app/plugins/cortex-debug/extension/support/openocd-helpers.tcl" -f /private/var/folders/ly/x8dq0h8x78d7j8l1xhctqb3r0000gn/T/arduino/sketches/7B83597618118A4EFCE532D04D575EA4/openocd.cfg
xPack Open On-Chip Debugger 0.12.0-01004-g9ea7f3d64-dirty (2023-01-30-17:05)
Licensed under GNU GPL v2
For bug reports, read
        http://openocd.org/doc/doxygen/bugs.html
CDRTOSConfigure
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
srst_only separate srst_nogate srst_open_drain connect_deassert_srst

Info : Listening on port 50001 for tcl connections
Info : Listening on port 50002 for telnet connections
Info : clock speed 1000 kHz
Info : STLINK V2J42S7 (API v2) VID:PID 0483:3748
Info : Target voltage: 3.248839
Info : [stm32f1x.cpu] Cortex-M3 r1p1 processor detected
Info : [stm32f1x.cpu] target has 6 breakpoints, 4 watchpoints
Info : starting gdb server for stm32f1x.cpu on 50000
Info : Listening on port 50000 for gdb connections
Info : accepting 'gdb' connection on tcp/50000
[stm32f1x.cpu] halted due to debug-request, current mode: Thread 
xPSR: 0x81000000 pc: 0x08003c0a msp: 0x20004fe0
Info : device id = 0x20036410
Info : flash size = 64 KiB
Error: timed out while waiting for target halted
Error: timed out while waiting for target halted
Error executing event gdb-flash-erase-start on target stm32f1x.cpu:
TARGET: stm32f1x.cpu - Not halted
Error: Target not halted
Error: failed erasing sectors 0 to 20
Error: flash_erase returned -304
Error: timed out while waiting for target halted
shutdown command invoked
[2024-09-11T18:25:24.430Z] SERVER CONSOLE DEBUG: onBackendConnect: gdb-server session closed
GDB server session ended. This terminal will be reused, waiting for next session to start...

**Comment by fpistm** [comment]
Hi @CJT1111 
is it a legacy STM32F103?
Did you enable the "optimize for debugging" in the menu?

On my side it works as expected.

**Comment by JingtaiChen** [comment]


**Reply by fpistm** [reply]
And the optimize for debug in the sketch menu?

**Comment by JingtaiChen** [comment]


**Reply by fpistm** [reply]
Maybe, edit the stm32f1x.cfg in the xpack open ocd folder and try to remove the `reset_config srst_nogate`

**Comment by JingtaiChen** [comment]


**Reply by fpistm** [reply]
No in` packages/STMicroelectronics/tools/xpack-openocd/0.12.0-3/openocd/scripts/target`

**Reply by fpistm** [reply]
OK. so try to connect the reset signal. 
After that I have no more idea except your mcu is not a legacy but a clone.

**Reply by JingtaiChen** [reply]
I switched to stlink to experiment, the chip is genuine, because it can even debug under cubeIDE

**Reply by JingtaiChen** [reply]
This stlink is a clone
